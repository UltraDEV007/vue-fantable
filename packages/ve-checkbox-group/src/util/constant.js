// prefix
export const PREFIX_CLS = 've-checkbox-group-'

// emit events
export const EMIT_EVENTS = {
  // on checked change
  ON_CHECKED_CHANGE: 'on-checked-change',
}

// comps name
export const COMPS_NAME = {
  VE_CHECKBOX: 'VeCheckbox',
  VE_CHECKBOX_GROUP: 'VeCheckboxGroup',
}
