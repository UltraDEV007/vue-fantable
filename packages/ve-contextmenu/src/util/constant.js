// prefix
export const PREFIX_CLS = 've-contextmenu-'

// comps name
export const COMPS_NAME = {
  VE_CONTEXTMENU: 'VeContextmenu',
}

// emit events
export const EMIT_EVENTS = {
  ON_NODE_CLICK: 'on-node-click',
}

// init data
export const INIT_DATA = {
  // PARENT_ID: 0,
  PARENT_DEEP: -1,
}

// contextmenu types
export const CONTEXTMENU_NODE_TYPES = {
  // Separator
  SEPARATOR: 'SEPARATOR',
}

// instance methods
export const INSTANCE_METHODS = {
  HIDE_CONTEXTMENU: 'hideContextmenu',
}
