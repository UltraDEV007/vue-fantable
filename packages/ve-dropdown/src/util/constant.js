// prefix
export const PREFIX_CLS = 've-dropdown-'

// emit events
export const EMIT_EVENTS = {
  FILTER_RESET: 'on-filter-reset',
  FILTER_CONFIRM: 'on-filter-confirm',
  VISIBLE_CHANGE: 'on-dropdown-visible-change',
  ITEM_SELECT_CHANGE: 'on-item-select-change',
}

// comps name
export const COMPS_NAME = {
  VE_DROPDOWN: 'VeDropdown',
}
