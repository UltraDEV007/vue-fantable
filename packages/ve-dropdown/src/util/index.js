/*
 * @clsName
 * @desc  get class name
 * @param {string} cls - class
 */
export function clsName(cls) {
  return 've-dropdown-' + cls
}
