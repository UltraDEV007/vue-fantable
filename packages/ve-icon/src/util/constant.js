// prefix
export const PREFIX_CLS = 've-icon-'

// emit events
export const EMIT_EVENTS = {}

// comps name
export const COMPS_NAME = {
  VE_ICON: 'VeIcon',
}
