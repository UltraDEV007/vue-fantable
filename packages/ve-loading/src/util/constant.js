// prefix
export const PREFIX_CLS = 've-loading-'

// emit events
export const EMIT_EVENTS = {}

// comps name
export const COMPS_NAME = {
  VE_LOADING: 'VeLoading',
  VE_LOADING_PLANE: 'VeLoadingPlane',
  /* VE_LOADING_CHASE: "VeLoadingChase", */
  VE_LOADING_BOUNCE: 'VeLoadingBounce',
  VE_LOADING_WAVE: 'VeLoadingWave',
  VE_LOADING_PULSE: 'VeLoadingPulse',
  VE_LOADING_FLOW: 'VeLoadingFlow',
  /* VE_LOADING_SWING: "VeLoadingSwing", */
  /*  VE_LOADING_CIRCLE: "VeLoadingCircle",
VE_LOADING_CIRCLE_FADE: "VeLoadingCircleFade", */
  VE_LOADING_GRID: 'VeLoadingGrid',
  VE_LOADING_FOLD: 'VeLoadingFold',
}

// spin name
export const SPIN_NAMES = {
  PLANE: 'plane',
  GRID: 'grid',
  WAVE: 'wave',
  FLOW: 'flow',
  BOUNCE: 'bounce',
  PULSE: 'pulse',
}
