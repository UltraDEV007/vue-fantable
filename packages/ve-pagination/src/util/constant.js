// prefix
export const PREFIX_CLS = 've-pagination-'

// locale comp name
export const LOCALE_COMP_NAME = 'pagination'

// emit events
export const EMIT_EVENTS = {
  PAGE_NUMBER_CHANGE: 'on-page-number-change',
  PAGE_SIZE_CHANGE: 'on-page-size-change',
}

// comps name
export const COMPS_NAME = {
  VE_PAGINATION: 'VePagination',
}
