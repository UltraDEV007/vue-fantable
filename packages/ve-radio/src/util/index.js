import { PREFIX_CLS } from './constant'

/*
 * @clsName
 * @desc  get class name
 * @param {string} cls - class
 */
export function clsName(cls) {
  return PREFIX_CLS + cls
}
