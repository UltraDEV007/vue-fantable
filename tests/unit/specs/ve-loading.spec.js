import { mount } from "@vue/test-utils";
import veLoading from "@/ve-loading";
import { later } from "../util";

describe("veLoading", () => {
    // spin names
    const SPIN_NAMES = {
        PLANE: "plane",
        GRID: "grid",
        WAVE: "wave",
        FLOW: "flow",
        BOUNCE: "bounce",
        PULSE: "pulse",
    };

    it("render normal", () => {
        const wrapper = mount({
            render() {
                return (
                    <div>
                        {Object.values(SPIN_NAMES).map((name) => {
                            return <div id={"loading-" + name} />;
                        })}
                    </div>
                );
            },
        });

        Object.values(SPIN_NAMES).forEach((spinName) => {
            veLoading({
                target: wrapper.vm.$el.querySelector(`#loading-${spinName}`),
                name: spinName,
            }).show();
        });

        expect(wrapper.html()).toMatchSnapshot();
    });

    it("fullscreen prop", async () => {
        const instance = veLoading({
            fullscreen: true,
            name: "bounce",
            lock: true,
        });

        await later();

        // body
        expect(
            document.body.classList.contains("ve-loading-parent-relative"),
        ).toBe(true);

        const loadingEl = document.querySelector(".ve-loading");

        expect(loadingEl).not.toBeNull();

        expect(loadingEl.classList.contains("ve-loading-hide")).toBe(true);

        instance.show();
        await later();
        expect(loadingEl.classList.contains("ve-loading-hide")).toBe(false);
    });

    it("lock prop", async () => {
        const instance = veLoading({
            fullscreen: true,
            name: "bounce",
            lock: true,
        });

        await later();

        const loadingEl = document.querySelector(".ve-loading");

        expect(loadingEl.classList.contains("ve-loading-fixed")).toBe(true);
    });

    it("tip prop", async () => {
        const wrapper = mount({
            render() {
                return (
                    <div>
                        <div class="loading-item"></div>
                    </div>
                );
            },
        });

        const tip = "loading...";
        const instance = veLoading({
            target: wrapper.vm.$el.querySelector(".loading-item"),
            name: "pulse",
            tip: tip,
        });

        expect(wrapper.find(".ve-loading-spin-tip").text()).toBe(tip);
    });

    it("overlayBackgroundColor prop", async () => {
        const wrapper = mount({
            render() {
                return (
                    <div>
                        <div class="loading-item"></div>
                    </div>
                );
            },
        });

        const bgColor = "rgba(255, 255, 255, 0.1)";
        const instance = veLoading({
            target: wrapper.vm.$el.querySelector(".loading-item"),
            name: "pulse",
            overlayBackgroundColor: bgColor,
        });

        expect(wrapper.find(".ve-loading").attributes("style")).toBe(
            `background-color: ${bgColor};`,
        );
    });

    it("show and close method", async () => {
        const wrapper = mount({
            render() {
                return (
                    <div>
                        <div class="loading-item"></div>
                    </div>
                );
            },
        });

        const instance = veLoading({
            target: wrapper.vm.$el.querySelector(".loading-item"),
            name: "pulse",
        });

        await later();
        expect(wrapper.find(".ve-loading-hide").exists()).toBe(true);

        instance.show();
        await later();
        expect(wrapper.find(".ve-loading-hide").exists()).toBe(false);

        instance.close();
        await later();
        expect(wrapper.find(".ve-loading-hide").exists()).toBe(true);
    });

    it("destory method", async () => {
        const wrapper = mount({
            render() {
                return (
                    <div>
                        <div class="loading-item"></div>
                    </div>
                );
            },
        });

        const instance = veLoading({
            target: wrapper.vm.$el.querySelector(".loading-item"),
            name: "pulse",
        });

        expect(wrapper.find(".ve-loading-spin").exists()).toBe(true);

        instance.destroy();
        await later();
        expect(wrapper.find(".ve-loading-spin").exists()).toBe(false);
    });
});
