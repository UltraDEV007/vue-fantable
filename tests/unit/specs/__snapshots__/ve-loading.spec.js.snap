// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`veLoading render normal 1`] = `
<div>
  <div id="loading-plane" class="ve-loading-parent-relative">
    <div class="ve-loading ve-loading-overlay ve-loading-hide" style="background-color: rgba(255, 255, 255, 0.5);">
      <div class="ve-loading-spin-container">
        <div class="ve-loading-spin">
          <div class="ve-loading-plane" style="width: 40px; height: 40px; background-color: rgb(24, 144, 255);"></div>
        </div>
        <div class="ve-loading-spin-tip" style="color: rgb(24, 144, 255);"></div>
      </div>
    </div>
  </div>
  <div id="loading-grid" class="ve-loading-parent-relative">
    <div class="ve-loading ve-loading-overlay ve-loading-hide" style="background-color: rgba(255, 255, 255, 0.5);">
      <div class="ve-loading-spin-container">
        <div class="ve-loading-spin">
          <div class="ve-loading-grid" style="width: 40px; height: 40px;">
            <div class="ve-loading-grid-cube" style="background-color: rgb(24, 144, 255);"></div>
            <div class="ve-loading-grid-cube" style="background-color: rgb(24, 144, 255);"></div>
            <div class="ve-loading-grid-cube" style="background-color: rgb(24, 144, 255);"></div>
            <div class="ve-loading-grid-cube" style="background-color: rgb(24, 144, 255);"></div>
            <div class="ve-loading-grid-cube" style="background-color: rgb(24, 144, 255);"></div>
            <div class="ve-loading-grid-cube" style="background-color: rgb(24, 144, 255);"></div>
            <div class="ve-loading-grid-cube" style="background-color: rgb(24, 144, 255);"></div>
            <div class="ve-loading-grid-cube" style="background-color: rgb(24, 144, 255);"></div>
            <div class="ve-loading-grid-cube" style="background-color: rgb(24, 144, 255);"></div>
          </div>
        </div>
        <div class="ve-loading-spin-tip" style="color: rgb(24, 144, 255);"></div>
      </div>
    </div>
  </div>
  <div id="loading-wave" class="ve-loading-parent-relative">
    <div class="ve-loading ve-loading-overlay ve-loading-hide" style="background-color: rgba(255, 255, 255, 0.5);">
      <div class="ve-loading-spin-container">
        <div class="ve-loading-spin">
          <div class="ve-loading-wave" style="width: 40px; height: 40px;">
            <div class="ve-loading-wave-rect" style="background-color: rgb(24, 144, 255);"></div>
            <div class="ve-loading-wave-rect" style="background-color: rgb(24, 144, 255);"></div>
            <div class="ve-loading-wave-rect" style="background-color: rgb(24, 144, 255);"></div>
            <div class="ve-loading-wave-rect" style="background-color: rgb(24, 144, 255);"></div>
            <div class="ve-loading-wave-rect" style="background-color: rgb(24, 144, 255);"></div>
          </div>
        </div>
        <div class="ve-loading-spin-tip" style="color: rgb(24, 144, 255);"></div>
      </div>
    </div>
  </div>
  <div id="loading-flow" class="ve-loading-parent-relative">
    <div class="ve-loading ve-loading-overlay ve-loading-hide" style="background-color: rgba(255, 255, 255, 0.5);">
      <div class="ve-loading-spin-container">
        <div class="ve-loading-spin">
          <div class="ve-loading-flow" style="width: 40px; height: 40px;">
            <div class="ve-loading-flow-dot" style="background-color: rgb(24, 144, 255);"></div>
            <div class="ve-loading-flow-dot" style="background-color: rgb(24, 144, 255);"></div>
            <div class="ve-loading-flow-dot" style="background-color: rgb(24, 144, 255);"></div>
          </div>
        </div>
        <div class="ve-loading-spin-tip" style="color: rgb(24, 144, 255);"></div>
      </div>
    </div>
  </div>
  <div id="loading-bounce" class="ve-loading-parent-relative">
    <div class="ve-loading ve-loading-overlay ve-loading-hide" style="background-color: rgba(255, 255, 255, 0.5);">
      <div class="ve-loading-spin-container">
        <div class="ve-loading-spin">
          <div class="ve-loading-bounce" style="width: 40px; height: 40px;">
            <div class="ve-loading-bounce-dot" style="background-color: rgb(24, 144, 255);"></div>
            <div class="ve-loading-bounce-dot" style="background-color: rgb(24, 144, 255);"></div>
          </div>
        </div>
        <div class="ve-loading-spin-tip" style="color: rgb(24, 144, 255);"></div>
      </div>
    </div>
  </div>
  <div id="loading-pulse" class="ve-loading-parent-relative">
    <div class="ve-loading ve-loading-overlay ve-loading-hide" style="background-color: rgba(255, 255, 255, 0.5);">
      <div class="ve-loading-spin-container">
        <div class="ve-loading-spin">
          <div class="ve-loading-pulse" style="width: 40px; height: 40px; background-color: rgb(24, 144, 255);"></div>
        </div>
        <div class="ve-loading-spin-tip" style="color: rgb(24, 144, 255);"></div>
      </div>
    </div>
  </div>
</div>
`;
